source/develop/release-management/features/storage/pass_discard_from_guest_to_underlying_storage.html.md

```diff
@@ -0,0 +1,234 @@
+
+#### The underlying storage
+
+##### Block storage
+The underlying storage must support *discard* calls in order for this to work.<br/>
+That's why when trying to enable *"Pass Discard"* for a VM disk, the engine first queries the disk's storage domain for its *discard* support.
+
+This is how we know if the storage supports *discard*:
+
+- Let *lunX* be a device and *dm-X* be its corresponding dm device for a natural number X. Then *lunX* is considered to ***support discard*** iff the value of */sys/block/dm-X/queue/discard_max_bytes* is bigger than 0 [^1].
+- A storage domain *supports discard* iff **all** of the luns that it is built from *support discard*. That means that if at least one of its luns doesn't *support discard*, then so is the storage domain.
+
+##### File storage
+Since there's another layer of file system in the middle, we can't check if the underlying storage supports *discard*. Hence, *Pass Discard* can be enabled for any disk on any file storage domain, and it will actually work only if the underlying file system supports it.
+
+#### The guest virtual machine
+In order to send a *discard* command from the guest, one can do it:
+
+- Manually - using the `fstrim` linux command.
+- Automatically on file removal (online discarding) - by adding the `discard` option to the list of the device's mount options. For example, in /etc/fstab:
+
+```properties
+/dev/mapper/rhel-root   /   ext4    defaults,discard    1   1
+# Here's the discard option -----------------^
+```
+
+Either way, these conditions regarding the *discard* command should be met:
+
+- It should be supported by the device's file system, *ext4* for example.
+- It should be supported by the guest's OS.
+
+#### The virtual machine's disk
+It is possible to enable *"Pass Discard"* for two disk storage types: **Images** and **Direct LUNs**.<br/>
+Regardless of the type, the disk interfaces that support *"Pass Discard"* are **VirtIO SCSI** and **IDE**.
+
+Note that for *Direct LUNs* with a *VirtIO SCSI* interface, enabling the *"Enable SCSI Pass-Through"* property already gives the disk *discard* support (among other things).<br/>
+We still need to decide about the relationship between these two fields - it should be discussed with the UX team (**<font color="red"> TBD </font>**).
+
+#### *Wipe After Delete* and *Pass Discard*
+
+##### Block storage
+When a disk's *Wipe After Delete* property is enabled, the engine guarantees that when the user removes the disk, it would be first wiped and only then removed.<br/>
+Since discarding a block in the storage means that it doesn't belong to the lv anymore, and since we cannot wipe this block before we discard it, if a disk's *Wipe After Delete* and *Pass Discard* properties were both enabled, we could have exposed some sensitive data as part of a new disk that was allocated with the same blocks to another user.<br/>
+Thus, *Wipe After Delete* and *Pass Discard* cannot, theoretically, be both enabled for the same disk.
```

Comment:
Since wiping is done by VDSM and discarding is done by the guest. Theoretically, qemu could implement something like this, but it's way out of scope for this feature.